### ISSUE TITLE
figure out docker

### ISSUE THREAD

**Issue by jaimeong** [jimmyjongs/passio#3]
do this first
https://docs.docker.com/engine/install/ubuntu/
https://docs.docker.com/get-started/

**Comment by jaimeong** [comment]
creating docker container
- in project folder, create a Dockerfile file (no extensions)
    - A Dockerfile is simply a text-based script of instructions that is used to create a container image
 - build using "docker build -t project_name ."
 - run using "docker run -dp port:port project_name"

**Comment by jaimeong** [comment]
updating docker container
- make code changes
- build 
- (if applicable) stop and remove previous version of container
    - docker ps
    - docker stop <container-id>
    - docker rm <container-id>

**Comment by jaimeong** [comment]
databases management 
- Volumes provide the ability to connect specific filesystem paths of the container back to the host machine. If a directory in the container is mounted, changes in that directory are also seen on the host machine. If we mount that same directory across container restarts, we’d see the same files.
- With the database being a single file, if we can persist that file on the host and make it available to the next container, it should be able to pick up where the last one left off. By creating a volume and attaching (often called “mounting”) it to the directory the data is stored in, we can persist the data. As our container writes to the todo.db file, it will be persisted to the host in the volume.

basically, create a database outside of container in question, then mount database to container. this persists the database between different version of the containerized app

- create volume
    -  docker volume create <dbname>
- start app container, but use -v to specify volume mount
    - docker run -dp port:port -v <dbfilepath>

**Comment by jaimeong** [comment]
https://golangdocs.com/golang-docker

**Comment by jaimeong** [comment]
Barebones containerized HTTP server created, see 
25a3b470e7afb6ece9230214f3ff80537d401668
